src/main/java/com/greenfox/tribesoflagopus/backend/service/BuildingService.java

```diff
@@ -0,0 +1,83 @@
+package com.greenfox.tribesoflagopus.backend.service;
+
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingLevelInputDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingListDto;
+import com.greenfox.tribesoflagopus.backend.model.entity.Building;
+import com.greenfox.tribesoflagopus.backend.model.entity.Kingdom;
+import com.greenfox.tribesoflagopus.backend.repository.BuildingRepository;
+import com.greenfox.tribesoflagopus.backend.repository.KingdomRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BuildingService {
+
+  @Autowired
+  BuildingRepository buildingRepository;
+
+  @Autowired
+  KingdomRepository kingdomRepository;
+
+  @Autowired
+  DtoService dtoService;
+
+  public BuildingListDto createBuildingList(long userId) {
+    List<Building> buildingsToConvertToDto = findKingdomByUserId(userId)
+        .getBuildings();
+    List<BuildingDto> buildingDtos = dtoService.convertFromBuildings(buildingsToConvertToDto);
+    BuildingListDto buildingsToReturn = BuildingListDto.builder()
+        .buildings(buildingDtos)
+        .build();
+    return buildingsToReturn;
+  }
+
+  public boolean validBuildingType(String inputBuildingType) {
+    if (inputBuildingType.equals("farm")
+        || inputBuildingType.equals("mine")
+        || inputBuildingType.equals("barrack")) {
+      return true;
+    }
+    return false;
+  }
+
+  public BuildingDto addNewBuilding(String type, long userId) {
+    Kingdom kingdomOfNewBuilding = findKingdomByUserId(userId);
+    createAndSaveNewBuilding(type, kingdomOfNewBuilding);
+    BuildingDto buildingDto = getTheNewestBuildingByTypeAndUser(kingdomOfNewBuilding, type);
```

Comment:
You can have one less DB query, if you return the new building in your `createAndSaveNewBuilding` method, and you just convert that here to DTO.